compiler/rustc_session/src/session.rs

```diff
@@ -1340,6 +1340,16 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
             }
         }
     }
+
+    if sess.opts.cg.soft_float {
```

Comment:
We should also mark the option as deprecated in it's description.
https://github.com/rust-lang/rust/blob/d6c8169c186ab16a3404cd0d0866674018e8a19e/compiler/rustc_session/src/options.rs#L1621-L1622